Make CUDA/NCCL plugins link against the existing ucc libraries.

Bart Oldeman, 2022-08-02
Mikael OEhman, 2023-06-16
diff -ur ucc-1.2.0.orig/src/components/mc/cuda/Makefile.am ucc-1.2.0/src/components/ec/cuda/Makefile.am
--- ucc-1.2.0.orig/src/components/mc/cuda/Makefile.am.orig	2023-06-16 12:56:53.205939925 +0200
+++ ucc-1.2.0/src/components/mc/cuda/Makefile.am	2023-06-16 13:02:21.716110609 +0200
@@ -14,7 +14,7 @@
 libucc_mc_cuda_la_CFLAGS   = $(BASE_CFLAGS)
 libucc_mc_cuda_la_LDFLAGS  = -version-info $(SOVERSION) --as-needed $(CUDA_LDFLAGS)
 libucc_mc_cuda_la_LIBADD   = $(CUDA_LIBS)                      \
-                             $(UCC_TOP_BUILDDIR)/src/libucc.la
+                             -lucc
 
 include $(top_srcdir)/config/module.am
 endif
diff -ur ucc-1.0.0.orig/src/components/tl/nccl/Makefile.am ucc-1.0.0/src/components/tl/nccl/Makefile.am
--- ucc-1.0.0.orig/src/components/tl/nccl/Makefile.am	2022-04-15 12:43:33.000000000 +0000
+++ ucc-1.0.0/src/components/tl/nccl/Makefile.am	2022-08-02 12:13:59.334795989 +0000
@@ -21,6 +21,6 @@
 libucc_tl_nccl_la_CPPFLAGS = $(AM_CPPFLAGS) $(BASE_CPPFLAGS) $(CUDA_CPPFLAGS) $(NCCL_CPPFLAGS)
 libucc_tl_nccl_la_CFLAGS   = $(BASE_CFLAGS)
 libucc_tl_nccl_la_LDFLAGS  = -version-info $(SOVERSION) --as-needed $(CUDA_LDFLAGS) $(NCCL_LDFLAGS)
-libucc_tl_nccl_la_LIBADD   = $(CUDA_LIBS) $(NCCL_LIBADD) $(UCC_TOP_BUILDDIR)/src/libucc.la
+libucc_tl_nccl_la_LIBADD   = $(CUDA_LIBS) $(NCCL_LIBADD) -lucc
 
 include $(top_srcdir)/config/module.am
